helios-testing/src/main/java/com/spotify/helios/testing/HeliosSoloDeployment.java

```diff
@@ -144,6 +168,94 @@
             .setUser(username)
             .setEndpoints("http://" + deploymentAddress)
             .build());
+
+    connectToWatchdog(dockerHost.address(), watchdogPort);
+
+    try {
+      start();
+    } catch (TimeoutException e) {
+      Throwables.propagate(e);
+    }
+  }
+
+  static synchronized HeliosDeployment getOrCreateHeliosSoloDeployment() {
+    if (soloDeployment == null) {
+      // TODO (dxia) remove checkForNewImages(). Set here to prevent using
+      // spotify/helios-solo:latest from docker hub
+      soloDeployment = HeliosSoloDeployment.fromEnv().checkForNewImages(false).build();
+    }
+    return soloDeployment;
+  }
+
+  private static void connectToWatchdog(final String host, final int port) {
+    final ExecutorService executorService = Executors.newSingleThreadExecutor(
+        new ThreadFactoryBuilder()
+            .setNameFormat("helios-solo-watchdog-%d")
+            .setDaemon(true).build());
+
+    log.info("Connecting to helios-solo watchdog at {}:{}", host, port);
+
+    executorService.submit(new Runnable() {
+      @Override
+      public void run() {
+        // 1. Connect
+        while (true) {
+          try {
+            final Socket s = new Socket();
+            s.connect(new InetSocketAddress(host, port));
+
+            // For whatever reason it seems like connections get "connected" even though there's
+            // really nothing on the other end -- to detect this send and recv some data.
+            final DataOutputStream writer = new DataOutputStream(s.getOutputStream());
+            final BufferedReader reader = new BufferedReader(
+                new InputStreamReader(s.getInputStream()));
+            writer.writeBytes("HELO\n");
+            writer.flush();
+            final String line = reader.readLine();
+            if (line.startsWith("HELO")) {
+              break;
+            } else {
+              s.close();
```

Comment:
why would the line not be HELO? would this be a case where you want to fail totally? Maybe at least log something.